crates/allocator/src/bump.rs

```diff
@@ -235,7 +267,251 @@ mod tests {
         let expected_limit = 3 * PAGE_SIZE;
         assert_eq!(inner.upper_limit, expected_limit);
 
-        let expected_alloc_start = 2 * PAGE_SIZE + std::mem::size_of::<u8>();
+        let expected_alloc_start = 2 * PAGE_SIZE + size_of::<u8>();
         assert_eq!(inner.next, expected_alloc_start);
     }
 }
+
+#[cfg(all(test, feature = "ink-fuzz-tests"))]
+mod fuzz_tests {
+    use super::*;
+    use quickcheck::{
+        quickcheck,
+        TestResult,
+    };
+    use std::mem::size_of;
+
+    #[quickcheck]
+    fn should_allocate_arbitrary_sized_bytes(n: usize) -> TestResult {
+        // If `n` is going to overflow we don't want to check it here (we'll check the overflow
+        // case in another test)
+        if n.checked_add(PAGE_SIZE - 1).is_none() {
+            return TestResult::discard()
+        }
+
+        let mut inner = InnerAlloc::new();
+
+        let layout = Layout::from_size_align(n, size_of::<usize>()).unwrap();
```

Comment:
I concur on the `required_pages` unwraps, but would really like to have an `expect` message for the `from_size_align` calls.